src/frontend/src/optimizer/plan_node/stream_sink.rs

```diff
@@ -315,6 +318,56 @@ impl StreamSink {
         let (pk, _) = derive_pk(input.clone(), user_order_by, &columns);
         let mut downstream_pk =
             Self::parse_downstream_pk(&columns, properties.get(DOWNSTREAM_PK_KEY))?;
+
+        {
+            // check for key encode TEXT
+            // if key encode is TEXT, we require the key to be a single column
+            // and the column should be of these types: varchar, bool, small int, int, big int
+            if let Some(format_desc) = &format_desc
+                && let Some(key_encode) = &format_desc.key_encode
+            {
+                if key_encode != &SinkEncode::Text {
+                    return Err(ErrorCode::SinkError(Box::new(Error::new(
+                        ErrorKind::InvalidInput,
+                        format!(
+                            "The key encode {:?} is not supported, only TEXT is supported.",
+                            &key_encode
+                        ),
+                    )))
+                    .into());
+                }
+                if downstream_pk.len() != 1 {
+                    return Err(ErrorCode::SinkError(Box::new(Error::new(
+                        ErrorKind::InvalidInput,
+                        format!(
+                            "The key encode is TEXT, but the primary key has {} columns. The key encode TEXT requires the primary key to be a single column.",
+                            downstream_pk.len()
+                        ),
+                    ))).into());
+                }
+
+                let key_col_catalog = columns.get(downstream_pk[0]).unwrap();
+                match &key_col_catalog.column_desc.data_type {
+                    DataType::Varchar
+                    | DataType::Boolean
+                    | DataType::Int16
+                    | DataType::Int32
+                    | DataType::Int64 => {}
+                    _ => {
```

Comment:
Please explicitly exhaust all data types here. It helps to you to think about all the data types, and also remains the other people to maintain this code when new data type is introduced.

And I think many data types are missing, such as `Decimal`, `Int256`, `Timestamp`, etc

Reply:
Refer to https://github.com/risingwavelabs/risingwave/pull/16377#discussion_r1591864960 for a list of all data types, and why we want to be conservative on some data types initially.